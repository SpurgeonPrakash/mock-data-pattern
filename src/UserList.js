import React from "react";

const UserList = ({ users }) => (
  <>
    <h2>Users</h2>
    <ul>
      {users.map(user => (
        <li key={user.id}>
          {user.name} {user.role === "Admin" && " - Admin"}
        </li>
      ))}
    </ul>
  </>
);

export default UserList;
