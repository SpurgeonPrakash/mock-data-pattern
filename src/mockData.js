export const users = [
  { id: 1, name: "Cory", role: "Admin" },
  { id: 2, name: "Megan", role: "User" }
];
