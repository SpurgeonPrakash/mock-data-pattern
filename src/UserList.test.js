import React from "react";
import UserList from "./UserList";
import { users } from "./mockData";
import renderer from "react-test-renderer";

it("renders user list with labels for admins", () => {
  const tree = renderer.create(<UserList users={users} />).toJSON();
  expect(tree).toMatchSnapshot();
});
