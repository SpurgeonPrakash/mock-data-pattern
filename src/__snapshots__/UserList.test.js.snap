// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user list with labels for admins 1`] = `
Array [
  <h2>
    Users
  </h2>,
  <ul>
    <li>
      Cory
       
       - Admin
    </li>
    <li>
      Megan
       
    </li>
  </ul>,
]
`;
